src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -3569,9 +3652,21 @@ boolean group(Context context, DeclarationList declList) throws ParserException
         }
 
         if (upcasts.contains(base.javaName)) {
+            // Base classes explicitly set as needing upcast in infoMap
             casts += upcast(type, base, true);
+            addDowncast(base.cppName, base);
+        } else if (polymorphicClasses.contains(base.javaName) && base.virtual) {
+            // In this case we know we need upcast
+            casts += upcast(type, base, false);
+            addDowncast(base.cppName, base);
         }
 
+        Set<Type> froms = downcasts.get(base.cppName);
+        if (froms != null)
+            for (Type t: froms) {
+                casts += downcast(type, t);
```

Comment:
Since this is actually a constructor, let's append to `constructors` instead?

Reply:
Ok, done.
I also added a check in case a constructor with the same signature has not been already added while parsing.